src/lib/storage/reference_column/reference_column_iterable.hpp

```diff
@@ -73,12 +73,16 @@ class ReferenceColumnIterable : public Iterable<ReferenceColumnIterable<T>> {
         return _value_from_value_column(*value_column, chunk_offset);
       }
 
-      if (auto dict_column = std::dynamic_pointer_cast<const DictionaryColumn<T>>(column)) {
+      if (auto dict_column = std::dynamic_pointer_cast<const DeprecatedDictionaryColumn<T>>(column)) {
         _dictionary_columns[chunk_id] = dict_column;
         return _value_from_dictionary_column(*dict_column, chunk_offset);
       }
 
-      Fail("Referenced column is neither value nor dictionary column.");
+      /**
+       * This is just a temporary solution to supporting encoded column type.
+       * It’s very slow and is going to be replaced very soon!
```

Comment:
It would be good to have a list of all the skeletons that we are putting in the closet with this PR.

Reply:
- [ ] visitor pattern should be revisited
- [ ] import / export updated
- [ ] deprecated dictionary column removed
- [ ] reference column updated (this is going to be my next PR)